src/machine.rs

```diff
@@ -532,6 +532,10 @@ pub struct MiriMachine<'tcx> {
     /// Needs to be queried by ptr_to_int, hence needs interior mutability.
     pub(crate) rng: RefCell<StdRng>,
 
+    /// The allocator used for the machine's `AllocBytes`.
+    #[cfg(target_os = "linux")]
+    pub(crate) allocator: Rc<RefCell<crate::alloc::isolated_alloc::IsolatedAlloc>>,
```

Comment:
This should be `Option<Rc<...>>` since it should only exist in native-libs mode.